src/fs/patch.ts

```diff
@@ -1,432 +1,107 @@
+    process.stderr.write(
+      // @ts-ignore
+      `[nexe] - DIRECTORIES ${JSON.stringify(Array.from(zipFs.listings.keys()), null, 4)}\n`
+    )
     log = (text: string) => {
-      setupManifest()
-      if (!loggedManifest) {
-        process.stderr.write('[nexe] - MANIFEST' + JSON.stringify(manifest, null, 4) + '\n')
-        process.stderr.write('[nexe] - DIRECTORIES' + JSON.stringify(directories, null, 4) + '\n')
-        loggedManifest = true
-      }
-      return process.stderr.write('[nexe] - ' + text + '\n')
-    }
-  }
-
-  const getKey = function getKey(filepath: string | Buffer | null): string {
-    if (Buffer.isBuffer(filepath)) {
-      filepath = filepath.toString()
-    }
-    if (!isString(filepath)) {
-      return notAFile
-    }
-    let key = path.resolve(baseDir, filepath)
-
-    return winPath(key)
-  }
-
-  const statTime = function () {
-    return {
-      atime: new Date(stat.atime),
-      mtime: new Date(stat.mtime),
-      ctime: new Date(stat.ctime),
-      birthtime: new Date(stat.birthtime),
-    }
-  }
-
-  let BigInt: Function
-  try {
-    BigInt = eval('BigInt')
-  } catch (ignored) {}
-
-  const minBlocks = Math.max(Math.ceil(stat.blksize / 512), 1)
-  const createStat = function (extensions: any, options: any) {
-    const stat = Object.assign(new fs.Stats(), binary.layout.stat, statTime(), extensions)
-    if ('size' in extensions) {
-      //Assume non adjustable allocation size for file system
-      stat.blocks = Math.ceil(stat.size / stat.blksize) * minBlocks
-    }
-    if (options && options.bigint && typeof BigInt !== 'undefined') {
-      for (const k in stat) {
-        if (Object.prototype.hasOwnProperty.call(stat, k) && typeof stat[k] === 'number') {
-          stat[k] = BigInt(stat[k])
-        }
-      }
-    }
-    return stat
-  }
-
-  const ownStat = function (filepath: any, options: any) {
-    setupManifest()
-    const key = getKey(filepath)
-    if (directories[key]) {
-      let mode = binary.layout.stat.mode
-      mode |= fs.constants.S_IFDIR
-      mode &= ~fs.constants.S_IFREG
-      return createStat({ mode, size: 0 }, options)
-    }
-    if (manifest[key]) {
-      return createStat({ size: manifest[key][1] }, options)
+      return process.stderr.write(`[nexe] - ${text}\n`)
     }
   }
-
-  const getStat = function (fn: string) {
-    return function stat(filepath: string | Buffer, options: any, callback: any) {
-      let stat: any
-      if (typeof options === 'function') {
-        callback = options
-        stat = ownStat(filepath, null)
-      } else {
-        stat = ownStat(filepath, options)
-      }
-      if (stat) {
-        process.nextTick(() => {
-          callback(null, stat)
-        })
-      } else {
-        return originalFsMethods[fn].apply(fs, arguments)
-      }
+  function internalModuleReadFile(this: any, original: any, ...args: any[]) {
+    log(`internalModuleReadFile ${args[0]}`)
+    try {
+      return fs.readFileSync(args[0], 'utf-8')
```

Comment:
`fs` in this context is the patched fs; so it will use the "check the zipfs and fallback to trying the realfs" logic implemented in SnapshotZipFS :-)

Re-reading though, using the `posixSnapshotZipFs` var directly makes the code clearer, so have pushed that in as a fixup commit (and would squash/fixup into 9d57b32)

Thanks for the suggestion :-)

(The integration test added in 3771c72 runs the built executable in an empty temp dir (it copies and spawns the binary in there) to test the executable without any of the source files being available :-) )